### ISSUE TITLE
netdata server/ cant setup second machine 

### ISSUE THREAD

**Issue by duhzon** [netdata/netdata#7152]
Hello there, im trying to do management in a network, the goal is make the server (192.168.1.100 in port 19999)  "see" all other 15 machines in one netdata interface. I tried start by installing netdata in every machine (First in 192.168.1.112) The problem is that  i cant run in it anything other than server, all other machine come along with the same issue (as you can see by output)!

Im using Fedora 26 (Server Edition)
My netdata is in version Netdata v1.18.1

Here is a list with some outputs:

```
2019-10-22 10:43:44: netdata INFO  : MAIN : SIGNAL: Not enabling reaper
2019-10-22 10:43:44: netdata ERROR : MAIN : LISTENER: IPv4 bind() on ip '192.168.1.112' port 19999, socktype 1 failed. (errno 98, Address already in use)
2019-10-22 10:43:44: netdata ERROR : MAIN : LISTENER: Cannot bind to ip '192.168.1.112', port 19999
2019-10-22 10:43:44: netdata FATAL : MAIN :LISTENER: Cannot listen on any API socket. Exiting... # : Success

2019-10-22 10:43:44: netdata INFO  : MAIN : EXIT: netdata prepares to exit with code 1...
2019-10-22 10:43:44: netdata INFO  : MAIN : EXIT: cleaning up the database...
2019-10-22 10:43:44: netdata INFO  : MAIN : Cleaning up database [0 hosts(s)]...
2019-10-22 10:43:44: netdata INFO  : MAIN : EXIT: all done - netdata is now exiting - bye bye.
```

And from:

```
netstat -nat | grep 19999
tcp        0      0 0.0.0.0:19999           0.0.0.0:*       LISTEN     
tcp6       0      0 :::19999                :::*                   LISTEN
```

Really thanks in advice guys!

**Comment by ilyam8** [comment]
Hi @duhzon 

> in one netdata interface

Do you mean [streaming](https://docs.netdata.cloud/streaming/)?

The error is self explanatory. To get the process use `p` flag and run the command as `root` user.


```
test@debian10:~$ sudo ss -natp | grep 19999
LISTEN  0       128                 0.0.0.0:19999                0.0.0.0:*       users:(("netdata",pid=13955,fd=4))
LISTEN  0       128                    [::]:19999                   [::]:*       users:(("netdata",pid=13955,fd=5))
```

**Comment by duhzon** [comment]
> Do you mean [streaming](https://docs.netdata.cloud/streaming/)?
Yeah for sure! im starting to study it. Before setting streaming i have to install netdata in every machine and it should run individually from de machine ip like (192.168.1.112:19999), right?

> The error is self explanatory. To get the process use `p` flag and run the command as `root` user.
Sorry my bad, there this comand was only to check, not the real problem. The real problem is that i cannot acess netdata from my browser: im in "ssh" into the machine 12 with ip 192.168.1.112 and i want run netdata from it (192.168.1.112:19999) but i only can do it with server 192.168.1.100:19999

**Comment by ilyam8** [comment]
> Before setting streaming i have to install netdata in every machine 

Yes. 

 - install netdata on every server
 - configure servers to stream to the master, restart netdata
 - after it you can browse all your netdata intances from master dashboard

I am not clear actually. Do you have problems with 1st step? You are not able to install netdata one some servers, are you?

**Comment by duhzon** [comment]
Thanks for the comment! 

> I am not clear actually. Do you have problems with 1st step? You are not able to install netdata one some servers, are you?

So my plan was to install netdata on every server. So i started with machine 12 (192.168.1.112) installation was successful, but i cant make it run on it!
If i try 192.168.1.15:19999 on my browser i got no response, when i try to acess netdata by terminal It come back some with error "Not enabling reaper" " Cannot bind to ip" "Cannot listen on any API socket"  and then exit de program (like i posted on thread as output).
I already tried change de port, bind to another ip and few other things on etc/netdata/netdata.confg)

So, how i cannot make it work in one machine i cant go through to step 2 and configure servers to stream to the master

Thanks in advice

**Comment by ilyam8** [comment]
> So i started with machine 12 (192.168.1.112) installation was successful

> , but i cant make it run on it!

Well, it is running right after install, you can access netdata via 192.168.1.112:19999

So:
 - you install netdata on 192.168.1.112 and installation was successful
 - netdata is running on 192.168.1.112
 - you cant access netdata from browser

?

**Comment by vlvkobal** [comment]
@duhzon, what is the output of the command `sudo ss -natp | grep 19999`, which Ilya mentioned above?
Do you have any firewall configured on 192.168.1.112?

**Comment by vlvkobal** [comment]
Also, please try invoking `curl http://localhost:19999/api/v1/info`. Do you get a valid response from Netdata?

**Comment by duhzon** [comment]
> Well, it is running right after install, you can access netdata via 192.168.1.112:19999
> 
> So:
> 
> * you install netdata on 192.168.1.112 and installation was successful
> * netdata is running on 192.168.1.112
> * you cant access netdata from browser
> 
> ?

I got no response when i type 192.168.1.112:19999 on browser, just a null page. Seems like the installation was sucessful but i cant even acess it from linux terminal, if i execute netdata by there got:

```
2019-10-28 09:09:11: netdata INFO  : MAIN : SIGNAL: Not enabling reaper
2019-10-28 09:09:11: netdata ERROR : MAIN : LISTENER: IPv4 bind() on ip '0.0.0.0' port 19999, socktype 1 failed. (errno 98, Address already in use)
2019-10-28 09:09:11: netdata ERROR : MAIN : LISTENER: Cannot bind to ip '0.0.0.0', port 19999
2019-10-28 09:09:11: netdata ERROR : MAIN : LISTENER: IPv6 bind() on ip '::' port 19999, socktype 1 failed. (errno 98, Address already in use)
2019-10-28 09:09:11: netdata ERROR : MAIN : LISTENER: Cannot bind to ip '::', port 19999
2019-10-28 09:09:11: netdata FATAL : MAIN :LISTENER: Cannot listen on any API socket. Exiting... # : Success

2019-10-28 09:09:11: netdata INFO  : MAIN : EXIT: netdata prepares to exit with code 1...
2019-10-28 09:09:11: netdata INFO  : MAIN : EXIT: cleaning up the database...
2019-10-28 09:09:11: netdata INFO  : MAIN : Cleaning up database [0 hosts(s)]...
2019-10-28 09:09:11: netdata INFO  : MAIN : EXIT: all done - netdata is now exiting - bye bye...

```

**Comment by duhzon** [comment]
> sudo ss -natp | grep 19999

Seems alright:
`LISTEN     0      128          *:19999                    *:*                   users:(("netdata",pid=893,fd=5))
LISTEN     0      128         :::19999                   :::*                   users:(("netdata",pid=893,fd=6))
`

**Comment by duhzon** [comment]
> Also, please try invoking `curl http://localhost:19999/api/v1/info`. Do you get a valid response from Netdata?

`{
	"version": "v1.18.1-20-nightly",
	"uid": "66a6a768-f0d9-11e9-b1ee-34e6d7fcc93e",
	"mirrored_hosts": [
		"lps12"
	],
	"alarms": {
		"normal": 87,
		"warning": 0,
		"critical": 0
	},
	"os_name": "Fedora",
	"os_id": "fedora",
	"os_id_like": "unknown",
	"os_version": "26 (Server Edition)",
	"os_version_id": "26",
	"os_detection": "/etc/os-release",
	"kernel_name": "Linux",
	"kernel_version": "4.16.11-100.fc26.x86_64",
	"architecture": "x86_64",
	"virtualization": "none",
	"virt_detection": "systemd-detect-virt",
	"container": "none",
	"container_detection": "systemd-detect-virt",
	"collectors": [
		{
			"plugin": "apps.plugin",
			"module": ""
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/diskstats"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/softirqs"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/uptime"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/meminfo"
		},
		{
			"plugin": "idlejitter.plugin",
			"module": ""
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/dev"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/stat/nf_conntrack"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/loadavg"
		},
		{
			"plugin": "proc.plugin",
			"module": "ipc"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/sockstat"
		},
		{
			"plugin": "statsd.plugin",
			"module": "stats"
		},
		{
			"plugin": "cgroups.plugin",
			"module": "stats"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/interrupts"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/snmp"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/stat"
		},
		{
			"plugin": "netdata",
			"module": "stats"
		},
		{
			"plugin": "diskspace.plugin",
			"module": ""
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/pagetypeinfo"
		},
		{
			"plugin": "web",
			"module": "stats"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/netstat"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/sys/kernel/random/entropy_avail"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/vmstat"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/snmp6"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/softnet_stat"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/sockstat6"
		}
	]
`
well, that i cant say but the response is above

**Comment by amoss** [comment]
Netdata is already running on 192.168.1.112, the installer finishes by starting it. This is why running `netdata` in the console gives you an error message. The output of curl on 192.168.1.112 shows that it is answering responses.

* Which machine are you running the browser on?
* Can you access the same information from that machine (by `curl http://192.168.1.112/api/v1/info`) ?
* How is the firewall configured on `192.168.1.112` ?

**Comment by duhzon** [comment]
> * ou running the



Thanks for the information.
-  Im running on bowser in my pc (im into ssh in a machine 12 with fixed ip 192.168.1.112). Funny thing is that i can acess 192.168.1.100 in any node, wich is the node 0 in networking, its acessible also by ssh, and from the main node i can ssh into any of 15 machines. For make things short, 192.168.1.100 is main node, i do a ssh into this by ip, from this i can acess any machine by ssh, answering the question im openning browser from my pc (without log out ssh) - Other machines dont have I/O devices only acessible by terminal ssh.

- Sure, i got this output: curl: (7) Failed to connect to 192.168.1.111 port 80: Connection refused
Already tried something with fw:
`sudo firewall-cmd --permanent --add-port=19999/tcp`

● This is firewalld status on 192.168.1.112:
 ```
firewalld.service - firewalld - dynamic firewall daemon
   Loaded: loaded (/usr/lib/systemd/system/firewalld.service; enabled; vendor preset: enabled)
   Active: active (running) since Mon 2019-10-28 09:45:04 EDT; 32min ago
     Docs: man:firewalld(1)
 Main PID: 823 (firewalld)
    Tasks: 2 (limit: 4915)
   CGroup: /system.slice/firewalld.service
           └─823 /usr/bin/python3 -Es /usr/sbin/firewalld --nofork --nopid
```

**Comment by amoss** [comment]
Sorry - I've put the wrong curl command in the previous message - it should be port 19999, not 80.
Could you try `curl http://192.168.1.112:19999/api/v1/info` on the machine with your browser?

It sounds as if your network is configured so that you have access to 192.168.1.100 from outside the subnet, and that your PC is outside of this subnet - is this correct?

**Comment by duhzon** [comment]
> `curl http://192.168.1.112:19999/api/v1/info`

`{
	"version": "v1.18.1-26-nightly",
	"uid": "66a6a768-f0d9-11e9-b1ee-34e6d7fcc93e",
	"mirrored_hosts": [
		"lps12"
	],
	"alarms": {
		"normal": 88,
		"warning": 0,
		"critical": 0
	},
	"os_name": "Fedora",
	"os_id": "fedora",
	"os_id_like": "unknown",
	"os_version": "26 (Server Edition)",
	"os_version_id": "26",
	"os_detection": "/etc/os-release",
	"kernel_name": "Linux",
	"kernel_version": "4.16.11-100.fc26.x86_64",
	"architecture": "x86_64",
	"virtualization": "none",
	"virt_detection": "systemd-detect-virt",
	"container": "none",
	"container_detection": "systemd-detect-virt",
	"collectors": [
		{
			"plugin": "apps.plugin",
			"module": ""
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/diskstats"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/softirqs"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/uptime"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/meminfo"
		},
		{
			"plugin": "idlejitter.plugin",
			"module": ""
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/dev"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/stat/nf_conntrack"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/loadavg"
		},
		{
			"plugin": "proc.plugin",
			"module": "ipc"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/sockstat"
		},
		{
			"plugin": "statsd.plugin",
			"module": "stats"
		},
		{
			"plugin": "cgroups.plugin",
			"module": "stats"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/interrupts"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/snmp"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/stat"
		},
		{
			"plugin": "netdata",
			"module": "stats"
		},
		{
			"plugin": "diskspace.plugin",
			"module": ""
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/pagetypeinfo"
		},
		{
			"plugin": "web",
			"module": "stats"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/netstat"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/rpc/nfs"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/sys/kernel/random/entropy_avail"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/vmstat"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/snmp6"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/softnet_stat"
		},
		{
			"plugin": "proc.plugin",
			"module": "/proc/net/sockstat6"
		}
	]
}`

Not only 192.168.1.100 there is other machine that i can browser ip and acess netdata in every node, but i dont know how and why only netdata on that machine works. When you say outside of subnet is with other ssid wifi? (im connect with the same ssid of node 0 router). I think cluster networking and subnet are all configured and working, im trying to make netdata running in each machine individually so then i'll use streaming for conecting machines on node 0.

**Comment by amoss** [comment]
Ok, let me check that I understand what you are doing:

* You have 16 machines setup as IP addresses 192.168.1.100 - 192.168.1.115 running on a single network.
* The browser that you are trying to connect with is running on 192.168.1.100
* The netdata server is running on 192.168.1.112
* The browser machine can see the netdata instance on port 19999 (because the curl command successfully retrieved the info from the API).
* The browser on the machine cannot access http://192.168.1.112:19999/

Could you show us how the browser fails? If it is chrome or firebox then you can right-click on the empty page, select "inspect element" and when the dev tools interface comes up click the network tab. Then press reload in the browser and take a screenshot that shows the browser window with the address bar visible and the the network accesses.

**Comment by duhzon** [comment]
Sure! its pretty much that! Netdata server is in 192.168.1.100 and its the main node, also i got 192.168.1.109 running into, i installed netdata and dont know why it worked only on that, i already check firewall for port in both and looks the same. Im trying to instal from 192.168.1.100 - 192.168.1.115 netdata but im stuck with 192.168.1.12 i tried unistall and re-install get uptade etc..
All what i got from network tab into inspect element from chrome is:

`Request URL: http://192.168.1.111:19999/
Referrer Policy: no-referrer-when-downgrade
Provisional headers are shown
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36`

![nd](https://user-images.githubusercontent.com/56880083/67703489-47673b80-f992-11e9-889b-2675c8df7172.jpeg)
It is in portuguese, so i will translate it mean 
its not possible to acess this site. The connect with 192.168.1.111 was recused.
search 192 168 111 19999 on Google
ERR_CONNECTION_REFUSED
ps: im using 192.168.1.111 because i decided try it on other machine i installed netdata, uptade looked at firewall and the problem still the same  as in 192.168.1.112

**Comment by duhzon** [comment]
Hey guys! I figured it out! in some machines i tried install by `bash <(curl -Ss https://my-netdata.io/kickstart.sh)` and installed only some essential package manually. `curl -Ss 'https://raw.githubusercontent.com/netdata/netdata-demo-site/master/install-required-packages.sh' >/tmp/kickstart.sh && bash /tmp/kickstart.sh -i netdata-all` solved my problem in 192.168.1.112. In other machine even with that package i failed in run it, but it was because firewall `sudo firewall-cmd --add-port=19999/tcp
sudo firewall-cmd --runtime-to-permanent` solved it! 
Now i have only 13 machine left to install netdata to begin work with stream to make node 0 seen all by netdata monitor.
Thank you guys your replies helped me to figure out and try every possibility, even try to install every package again,

**Comment by amoss** [comment]
Great to hear. Good luck with your install and please feel free to reopen the issue if you run into the same problem again.
